app/models/concerns/profile_policy_association.rb

```diff
@@ -14,6 +14,14 @@ module ProfilePolicyAssociation
              to: :policy_object, allow_nil: true
     validate :no_duplicate_policy_types, on: :create
 
+    def policy_profile
+      policy_object&.initial_profile || self
```

Comment:
What's the rationale on returning `self`? It does not seem to align with previous policy usage on GQL and neither with the new ways. There might be a case of an unexpected behavior.

Reply:
For external policies, the single profile involved is the policy itself. The UI shows a similar logic - every report has some policy, be it external or internal; in the same way, this attribute will always return some profile ID. I'm willing to change it for consistency to return `nil` for profiles without a policy.

Reply:
External profiles previously did not get `policy` on the GQL (only if they where created after an internal profile creation). The problem of the OR was that it would return `self` if `initial_profile` yields `nil`. A policy not having an initial profile would already a bad state.

Thank you for the change.